sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/SearchableFieldProperty.java

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search.documents.indexes;
+
+import com.azure.search.documents.models.AnalyzerName;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * The annotation is to indicate whether the field is a searchable field. The boolean field of isSearchable
+ * defaults to true if use the annotation.
+ */
+@Target({ElementType.FIELD})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface SearchableFieldProperty {
```

Comment:
Missing `name`. The name must be overridable. It can default to the property name, but that isn't always desirable. Same for all other attribute classes below.

Reply:
I have added name for Searchable, Simple attributes.

Reply:
This leaves a problem for complex fields since we don't have an attribute.

Maybe we need to punt field renaming to preview 4. Come to think of it, in .NET different attributes (from Json.NET, though in track 2 we'd use the built-in ones) could rename it. You could do the same with your JSON serialization framework (I forget the name off hand).

/cc @brjohnstmsft

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/10647

This is a summery issue for all open questions not going to address for this PR.